In-app marketplace redesign phase 1

andfinally opened the pull request:
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

The Extend team wants to refresh the design of the in-app marketplace and better align it with the marketplace on WooCommerce.com. (Figma: seb2o1DExPxoh5d53ftJif-fi-812%3A1.)

This PR bundles the changes from several branches for phase 1 of the in-app marketplace rebuild. We're rendering the current "Browse Extensions" and "WooCommerce.com Subscriptions" tabs of the "Extensions" page as two separate pages, "Marketplace" and "My Subscriptions". "My Subscriptons" is pretty much unchanged, apart from that. "Marketplace" has a new header and restyled product cards.

We're hoping to build on these changes in phase 2 of the project, in which we'll rebuild the "Featured" section, add more data to the product cards, add a WooCommerce.com Helper connection element and other improvements.

#### PRs in this branch:

* [Split the Extensions page in WP Admin and add respective menu items](https://github.com/woocommerce/woocommerce/pull/30380)
* [New marketplace addons header](https://github.com/woocommerce/woocommerce/pull/30389)
* [Addressing PHPCS issues in Helper view file](https://github.com/woocommerce/woocommerce/pull/30405)
* [Add product card styling](https://github.com/woocommerce/woocommerce/pull/30410)
* [Layout tweaks](https://github.com/woocommerce/woocommerce/pull/30478)
* [Restyle Marketplace tab bar](https://github.com/woocommerce/woocommerce/pull/30464)
* [Update My Subscriptions page title](https://github.com/woocommerce/woocommerce/pull/30445)
* [In-app marketplace style tweaks](https://github.com/woocommerce/woocommerce/pull/30497)
* [Add Tracks event for new My Subscriptions page](https://github.com/woocommerce/woocommerce/pull/30488)
* [Display notices properly in Marketplace page](https://github.com/woocommerce/woocommerce/pull/30478)
* [In-app marketplace style tweaks](https://github.com/woocommerce/woocommerce/pull/30497)
* [Marketplace mobile layout fixes](https://github.com/woocommerce/woocommerce/pull/30498)
* [Add mobile section menu to marketplace](https://github.com/woocommerce/woocommerce/pull/30499)

Related WC Admin PR: [Split the Extensions page into Marketplace and My Subscriptions](https://github.com/woocommerce/woocommerce-admin/pull/7471)

### How to test the changes in this Pull Request:

These are general steps for the overall change – please see the constituent PRs if you want to make more detailed checks.

1. Check out this branch.
2. Notice the WooCommerce > Extensions menu item has been replaced by WooCommerce > Marketplace and WooCommerce > My Subscriptions.
3. Please disregard the "Extensions" title in the breadcrumb on these two pages – we're addressing that in a WooCommerce Admin [PR](https://github.com/woocommerce/woocommerce-admin/pull/7471).
4. View the My Subscriptions page. You should see your store's connection to WooCommerce.com as usual, or connect it.
5. View the Marketplace page. It should appear like the designs on all viewports. You should be able to browse the different sections of the marketplace and make searches as normal. Clicking on a product's "View details" button should take you to the product page. Promotions should appear as normal in the Featured section and (depending on your locale and plugin setup) in the Shipping and Payments sections.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

Enhancement - Split the "Extensions" page into two pages, "Marketplace" and "My Subscriptions". Apply new design to Marketplace.

### Screenshots

<img width="500" src="https://user-images.githubusercontent.com/1647564/129351097-5f61c74d-171a-4361-a0a8-869ae0cb8396.png">

<img width="250" src="https://user-images.githubusercontent.com/1647564/129363255-f94f18e8-9385-4d0e-9e80-30bab63b4be2.png">

octaedro committed:
Bump woocommerce-admin version to 2.5.0-rc.1

claudiosanches committed:
Fixed changelog entry type for 5.6.0

claudiosanches committed:
Added missing entry

Konamiman committed:
Add logging to the GH action for assigning a milestone to merged PRs

If the GH API call to assign the milestone to a merged PR returns an
error, the data returned from the call will be dumped to the console,
and thus will be readable by looking at the action execution log.

roykho committed:
Revert the use of prop here as we want to remove the attribute completely closes #30282

madeincosmos committed:
Split Extensions page into 2 menu items.

madeincosmos committed:
Update e2e tests.

Konamiman committed:
Use 'pull_request_target' in the automatic milestoning action

The original event used, 'pull_request', runs in the context of the
pull request. In the case of pull requests created from forks this means
that a read-only GitHub token is used for API calls, and thus the
call used to assign the milestone fails.

The fix is using the 'pull_request_target' event, which runs in the
context of the base branch, and thus with a read-write token. See:
https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_target

Also, added a warning message when no milestone without a release
branch is found and thus the newest existing milestone is assigned
(this is a fallback that should normally not happen).

madeincosmos committed:
Change menu item name to Marketplace.

madeincosmos committed:
Change My Subscriptions page header.

Konamiman committed:
Fix option names used by the product attributes lookup table feature

The feature was using option names prefixed with
"woocommerce_attribute_lookup__". That double underscore breaks WP
standards for option names. The option names have been changed
so that they use a single underscore instead.

kloon committed:
New marketplace addons header

kloon committed:
New marketplace addons header

andfinally committed:
Style tweaks. Added mobile styles. Using default font-family. Removed letter-spacing. Using flexbox to centre the content in the header.

andfinally committed:
Style tweaks. Added mobile styles. Using default font-family. Removed letter-spacing. Using flexbox to centre the content in the header.

andfinally committed:
Style tweaks. Using flexbox. Using default font. Removed letter-spacing. Added mobile style.

andfinally committed:
Style tweaks. Using flexbox. Using default font. Removed letter-spacing. Added mobile style.

andfinally committed:
Merge branch 'add/10864-marketplace-purple-search-background' of https://github.com/woocommerce/woocommerce into add/10864-marketplace-purple-search-background

andfinally committed:
Giving 100% width to purple header.

andfinally committed:
Giving 100% width to purple header.

andfinally committed:
Merge pull request #30389 from woocommerce/add/10864-marketplace-purple-search-background

New marketplace addons header

roykho committed:
Update assets/js/admin/quick-edit.js

Co-authored-by: jonathansadowski <jonathansadowski@users.noreply.github.com>

roykho committed:
Add task instructions for the PR reviewer

andfinally committed:
Addressing PHPCS issues.

- Added file doc comment.
- Escaping direct output and translated output.
- Added translators comments.
- Renaming `$action` vars to `$subscription_action`, so they don't override the global of the same name.

andfinally committed:
Merge branch 'update/marketplace-1' into fix/admin-helper-views-main-phpcs

madeincosmos committed:
Add product card styling.

andfinally committed:
Merge pull request #30405 from woocommerce/fix/admin-helper-views-main-phpcs

Addressing PHPCS issues in Helper view file

madeincosmos committed:
Align product cards on the Extensions page.

madeincosmos committed:
Cleanup unused functions and files.

madeincosmos committed:
Fix phpcs issues.

andfinally committed:
Merge branch 'update/marketplace-1' into update/split-extensions-pages

madeincosmos committed:
Merge branch 'update/split-extensions-pages' of github.com:woocommerce/woocommerce into update/split-extensions-pages

madeincosmos committed:
Update Marketplace and My Subscriptions page titles.

andfinally committed:
Merge branch 'trunk' into update/marketplace-1

andfinally committed:
Correcting escaping in the subscriptions list in `html-main.php` – `esc_html` was escaping a `strong` tag in the renewal message.

madeincosmos committed:
Merge pull request #30380 from woocommerce/update/split-extensions-pages

Split the Extensions page in WP Admin and add respective menu items

madeincosmos committed:
Update document title for My Subscriptions.

andfinally committed:
Merge pull request #30445 from woocommerce/update/marketplace-page-title

Update My Subscriptions page title

andfinally committed:
Merge branch 'update/marketplace-1' into update/styling-product-cards

andfinally committed:
CSS and layout tweaks.

- Changed `wc_addons_wrap` to `wc-addons-wrap` to adhere to stylelint class rule.
- Removed `max-width` from `wc-addons-wrap` – we need the purple header to stretch to full width. Added new `wc-subscriptions-wrap` class to give the `max-width` to the My Subscriptions page.
- Changed width of product cards to use `calc`, so that any trailing odd-numbered card has the same width as the others.
- Fixed some stylelint issues. Moved style blocks to address `no-descending-specificity` warnings. Remove quotes from URLs and font names that don't need them.
- Gave `#screen-meta-links` absolute position on the marketplace page, so it overlays the full-width purple header.

andfinally committed:
Addressing PHPCS errors.

- Deleted precision indentation on lines 72 and 80.
- Deleted unnecessary `echo` from before `esc_html_e` cals.

madeincosmos committed:
Fix box shadow and margins.

madeincosmos committed:
Update current item highlight.

madeincosmos committed:
Add product list margin on larger screens.

andfinally committed:
Merge branch 'trunk' into update/marketplace-1

# Conflicts:
#	tests/e2e/core-tests/specs/merchant/wp-admin-extensions-connect-wccom.test.js

andfinally committed:
Added filter to `WC_Admin` which adds special classes `woocommerce_page_wc-subscriptions` and `woocommerce_page_wc-marketplace` to the new split pages.

andfinally committed:
Layout tweaks. Addressing positioning and styling of systems notices like the regenerating thumbnails notice, which should appear above the purple header. Changed `wc_addons_wrap` class to kebab case to address stylelint issue with class selector.

madeincosmos committed:
Merge pull request #30410 from woocommerce/update/styling-product-cards

Add product card styling.

andfinally committed:
Layout tweaks. Changed new page classes to kebab case to avoid future stylelint selector warnings. Styling notices when they appear above the purple header.

madeincosmos committed:
Rebase with feature branch.

madeincosmos committed:
Add mobile styles for top bar.

madeincosmos committed:
Add menu scroll on mobile.

andfinally committed:
Merge branch 'update/marketplace-1' into update/layout-tweaks

# Conflicts:
#	assets/css/admin.scss
#	includes/admin/views/html-admin-page-addons.php

andfinally committed:
Simplifying styling of notices on Marketplace and My Subscriptions pages.

- Added `.wp-header-end` div on Marketplace page, so WordPress will insert `admin_notices` there.
- Added top margin on notice on My Subscriptions page to keep it clear of Help tab.

andfinally committed:
Removed `wrap` class from `.wc-addons-wrap` div on Marketplace page, as we now have a way of forcing the notices to appear at a particular place. Removing styles associated with the previous classes.

andfinally committed:
Fixing PHPCS issue - return parameter comment.

andfinally committed:
Fixing PHPCS issue - added return parameter comment.

andfinally committed:
Merge branch 'trunk' into update/marketplace-1

andfinally committed:
Merge branch 'update/marketplace-1' into update/layout-tweaks

andfinally committed:
Fixing PHPCS issue - added parameter comment.

madeincosmos committed:
Center the menu on larger screens.

andfinally committed:
Updated `track_extensions_page` to record new Tracks event `wcadmin_subscriptions_view` if the user views the My Subscriptions page.

andfinally committed:
Merge pull request #30478 from woocommerce/update/layout-tweaks

Update/layout tweaks

madeincosmos committed:
Fix menu shadow and margins.

andfinally committed:
Merge branch 'update/marketplace-1' into update/restyle-tab-bar

madeincosmos committed:
Merge pull request #30464 from woocommerce/update/restyle-tab-bar

Restyle Marketplace tab bar

andfinally committed:
Merge pull request #30488 from woocommerce/update/marketplace-page-load-track

Update/marketplace page load track

andfinally committed:
Merge branch 'trunk' into update/marketplace-1

andfinally committed:
Small style tweaks.

- Added top margin to `.addons-button` to ensure there's a gap above it when promos are squished to a narrower width.
- Changed wording and style of `Search results for` label at the top of extensions search results, to conform to new design.

madeincosmos committed:
Fix product card width on mobile.

andfinally committed:
Restored classname in style block, accidentally deleted.

madeincosmos committed:
Initial dropdown menu styling.

madeincosmos committed:
Add dropdown arrow.

madeincosmos committed:
Fix linter errors.

andfinally committed:
Addressing PHPCS issues.

andfinally committed:
Reduced left padding of `.marketplace-header` on smallest screens, so text in it lines up with the breadcrumb menu and section dropdown link.

andfinally committed:
Merge pull request #30497 from woocommerce/update/marketplace-style-tweaks

In-app marketplace style tweaks

andfinally committed:
Merge branch 'update/marketplace-1' into update/marketplace-mobile-fixes

andfinally committed:
Merge branch 'update/marketplace-1' into update/marketplace-mobile-fixes

# Conflicts:
#	assets/css/admin.scss

andfinally committed:
Added mobile style section menu to marketplace.

andfinally committed:
Merge pull request #30498 from woocommerce/update/marketplace-mobile-fixes

Marketplace mobile layout fixes

andfinally committed:
Merge branch 'update/marketplace-1' into update/marketplace-mobile-menu

andfinally committed:
Increased width of tablet/desktop menu. Decreased top and bottom padding of tablet/desktop menu items.

andfinally committed:
Added logic to close marketplace section menu if user clicks outside it.

andfinally committed:
Merge pull request #30499 from woocommerce/update/marketplace-mobile-menu

Add mobile section menu to marketplace

vedanshujain commented on assets/css/admin.scss:
```diff
@@ -506,87 +655,114 @@
+		max-width: 1032px;
+		overflow: hidden;
 
+		/* stylelint-disable no-descending-specificity */
```
let's add a comment specifying why this stylelint was disabled?

andfinally replied:
I've removed the `stylelint-disable` comments in https://github.com/woocommerce/woocommerce/pull/30391/commits/28881769443813d8c5e281fdba8be3f0fb002cdc.

vedanshujain commented on includes/admin/class-wc-admin-menus.php:
```diff
@@ -36,6 +36,8 @@ public function __construct() {
 		add_filter( 'menu_order', array( $this, 'menu_order' ) );
 		add_filter( 'custom_menu_order', array( $this, 'custom_menu_order' ) );
 		add_filter( 'set-screen-option', array( $this, 'set_screen_option' ), 10, 3 );
+		add_filter( 'parent_file', array( $this, 'update_menu_highlight' ) );
```
can we hook into `submenu_file` filter instead? that way we won't have to override `submenu_file` global inside this function and this will be more resilient.

andfinally replied:
Thanks! Done in https://github.com/woocommerce/woocommerce/pull/30391/commits/28881769443813d8c5e281fdba8be3f0fb002cdc.

vedanshujain commented on includes/admin/helper/views/html-main.php:
```diff
@@ -32,7 +59,7 @@
 			?>
 			<li>
-				<a <?php echo $class_html; ?> href="<?php echo esc_url( $url ); ?>">
+				<a <?php echo esc_attr( $class_html ); ?> href="<?php echo esc_url( $url ); ?>">
```
this should be `esc_html`

andfinally replied:
Escaping fixed in https://github.com/woocommerce/woocommerce/pull/30391/commits/af9e2b17774d9d8f6a532d54c2e62cdf2bba466a, thanks for spotting!

vedanshujain commented on includes/admin/views/html-admin-page-addons.php:
```diff
@@ -11,124 +11,148 @@
+
+	<div class="wrap">
+		<div class="marketplace-content-wrapper">
+			<?php if ( isset( $_GET['search'] ) ) : // phpcs:ignore WordPress.Security.NonceVerification.Recommended ?>
```
we should pass the `$search` param from the calling code instead, and use it here. that way we won't have to do phpcs:ignore three times in this template.

andfinally replied:
Addressed in https://github.com/woocommerce/woocommerce/pull/30391/commits/af9e2b17774d9d8f6a532d54c2e62cdf2bba466a.

vedanshujain commented on includes/admin/views/html-admin-page-addons.php:
```diff
@@ -11,124 +11,148 @@
+								<?php if ( ! empty( $addon->image ) ) : ?>
+									<span class="product-img-wrap"><img src="<?php echo esc_url( $addon->image ); ?>" /></span>
+								<?php endif; ?>
+								<a href="<?php echo esc_attr( WC_Admin_Addons::add_in_app_purchase_url_params( $addon->link ) ); ?>">
```
this should `esc_url`

andfinally replied:
Done in https://github.com/woocommerce/woocommerce/pull/30391/commits/af9e2b17774d9d8f6a532d54c2e62cdf2bba466a.

vedanshujain commented on includes/admin/views/html-admin-page-addons.php:
```diff
@@ -11,124 +11,148 @@
+									<span class="price"><?php echo wp_kses_post( $addon->price ); ?></span>
+									<span class="price_suffix"><?php esc_html_e( 'per year', 'woocommerce' ); ?></span>
+								<?php endif; ?>
+								<a class="button" href="<?php echo esc_attr( WC_Admin_Addons::add_in_app_purchase_url_params( $addon->link ) ); ?>">
```
`esc_url` here as well

andfinally replied:
Done in https://github.com/woocommerce/woocommerce/pull/30391/commits/af9e2b17774d9d8f6a532d54c2e62cdf2bba466a.

vedanshujain reviewed (changes requested):
Thanks for the changes, and appreciate all the additional escaping in old code as well. Great work!

I have added some code comments which should be addressed. Other than those I noticed following things in my testing:

1. "My Subscription" menu item does not appear for wp.com business plans and this could be a blocker. It does appear everywhere else I tested though.
2. Maybe we want more aggressive z-index in the category dropdown, currently it appears behind any notice in the shop, see:
<img width="1301" alt="Screenshot 2021-08-17 at 2 11 31 PM" src="https://user-images.githubusercontent.com/7571618/129694174-061d9933-ca00-44e4-92c5-55fce0bb4d30.png">

Because of this it's very hard to navigate between various categories.

madeincosmos committed:
Fix z-index for Extensions menu.

madeincosmos commented:
> Maybe we want more aggressive z-index in the category dropdown, currently it appears behind any notice in the shop

Thanks, this is fixed now.

andfinally committed:
Addressing feedback.

- Deleted `stylelint-disable` comments.
- Changed hook for `update_menu_highlight` from `parent_file` to `submenu_file`, so we don't need to access `$submenu_file` global.

andfinally committed:
Addressing feedback.

- Using `$search` variable instead of `$_GET['search']` to minimise `phpcs:ignore` comments.
- Using correct escaping functions.

andfinally committed:
Fixing linter errors.

- Added Doc comment for parameter "$submenu_file".
- Removed space between `empty` and opening parenthesis.

andfinally commented:
Thanks @vedanshujain, apologies for the big PR! We've hopefully addressed the issues – Maria just had a question about the WP.com issue.

vedanshujain commented on includes/admin/helper/views/html-section-nav.php:
```diff
@@ -1,19 +0,0 @@
-<?php
```
Sorry, I missed this earlier, I don't think we can remove a file because of backward compatibility policies. Can we deprecate this instead?

madeincosmos replied:
Thanks, I restored the file and added `@deprecated` tag.

vedanshujain commented on includes/admin/views/html-admin-page-addons.php:
```diff
@@ -11,124 +11,148 @@
 				name="search"
-				value="<?php echo esc_attr( isset( $_GET['search'] ) ? sanitize_text_field( wp_unslash( $_GET['search'] ) ) : '' ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended ?>"
-				placeholder="<?php esc_attr_e( 'Enter a search term and press enter', 'woocommerce' ); ?>">
+				value="<?php echo esc_attr( ! empty( $search ) ? sanitize_text_field( wp_unslash( $search ) ) : '' ); ?>"
```
I think we would also need to add `$search` variable in `uses` doc comment in file `includes/admin/class-wc-admin-addons.php:730`.

madeincosmos replied:
Added it now :)

vedanshujain reviewed (commented):
Great work! I have added a couple of minor comments but look good otherwise.

madeincosmos committed:
Restore and deprecate the helper nav file.

madeincosmos committed:
Add missing doc parameter.

andfinally commented:
Hi @vedanshujain, Maria has addressed the further issues. You're seeing the issue with the navigation on Atomic because we didn't update the new nav menu in WC Admin. Maria's now got a [WC Admin PR](https://github.com/woocommerce/woocommerce-admin/pull/7529) for that change.

<img width="500" src="https://user-images.githubusercontent.com/1647564/129926599-7f9703d0-5a45-4e64-95fc-638538d414af.png">

andfinally committed:
Merge branch 'trunk' into update/marketplace-1

vedanshujain reviewed (dismissed):
LGTM! Thanks for the changes, e2e tests are failing but it does not look related to this PR. Going to run them again before merging.

andfinally commented:
🎉 Woohoo! Thanks so much for your help Vedanshu!

andfinally commented:
https://github.com/woocommerce/woocommerce/pull/30535 has all the changes from this branch except one, an [ESLint comment](https://github.com/woocommerce/woocommerce/pull/30391/files#diff-f66b4209f2dd815b841b45c2f3f8a6eff8d19849112111a223e09d532e7fc49eR1) at the top of `tests/e2e/core-tests/specs/merchant/wp-admin-extensions-connect-wccom.test.js`).  That PR passes the e2e tests. So I'd say it's safe to merge this one @vedanshujain. Alternatively, I could prepare https://github.com/woocommerce/woocommerce/pull/30535 to merge?

andfinally committed:
Deleted unneeded ESLint comment.

vedanshujain reviewed (approved):
LGTM!
